variationGenerator.js

```diff
@@ -117,11 +187,19 @@ function removeGroup(at, board) {
     );
 }
 
-function getNeighbors(cords) {
-    return [
-        [cords[0], cords[1] - 1],
-        [cords[0], cords[1] + 1],
-        [cords[0] + 1, cords[1]],
-        [cords[0] - 1, cords[1]]
-    ].filter(([x, y]) => x > 0 && x < 20 && y > 0 && y < 20);
+const NEIGHBORS = [];
+for (let x = 1; x <= 19; x++) {

```

Comment:
It's interesting that you found this optimization actually makes difference in runtime, since the original implementation of `getNeighbors` also has constant complexity (checking exactly 4 elements each run), so this is only a constant optimization.
I think the idea is good but the not the direction, maybe memoizing `getLiberties` or `makeMove` will have better impact on runtime?